### PR TITLE
Use AccuWeather for city lookup

### PR DESCRIPTION
**Pull request by jamesgorrie**
Co-authored-by: Max Duval <max.duval@theguardian.com>

## What does this change?
This removes the need to use a CSV lookup for a city as Accuweather support ISO country and region codes that we now get from Fastly.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots
Screenshot 2022-05-23 at 13 12 26

Screenshot 2022-05-23 at 13 13 58

Screenshot 2022-05-23 at 13 23 28

Alternative to #25031

### CODE REVIEW

File: onward/app/weather/controllers/LocationsController.scala [hunk 1]
```diff
@@ -34,44 +34,27 @@ class LocationsController(weatherApi: WeatherApi, val controllerComponents: Cont
       def getEncodedHeader(key: String) =
         request.headers.get(key).map(java.net.URLDecoder.decode(_, "latin1"))
 
+      val maybeCountry = getEncodedHeader(CountryHeader).filter(_.nonEmpty)
       val maybeCity = getEncodedHeader(CityHeader).filter(_.nonEmpty)
       val maybeRegion = getEncodedHeader(RegionHeader).filter(_.nonEmpty)
```
Comment (mxdvl) [inline onward/app/weather/controllers/LocationsController.scala:39]: Why reorder these?

File: onward/app/weather/controllers/LocationsController.scala [hunk 2]
```diff
@@ -34,44 +34,27 @@ class LocationsController(weatherApi: WeatherApi, val controllerComponents: Cont
...
+      (maybeCountry, maybeCity) match {
+        case (Some(countryCode), Some(city)) =>
+          weatherApi.searchForCity(countryCode, city, maybeRegion) map { locations =>
+            val cities = CityResponse.fromLocationResponses(locations.filter(_.Country.ID == countryCode).toList)
+            cities.headOption.fold {
+              log.warn(s"Could not find $countryCode, $city")
+              Cached(CacheTime.NotFound)(JsonNotFound())
+            } { weatherCity =>
+              log.info(s"Matched $countryCode, $city, $maybeRegion to ${weatherCity.id}")
+              // We do this as accuwweather writes "New York, New York" if no region is specified, where as we
```
Comment (mxdvl) [inline onward/app/weather/controllers/LocationsController.scala:50]:
```suggestion
              // We do this as AccuWeather writes "New York, New York" if no region is specified, where as we
```

**Review by mxdvl — APPROVED** [review]
Great simplification!

We could still add some tests, possibly at least for all the cities where we have an office.

PS. That’s a great ratio `+26 -353,352`

File: onward/app/weather/controllers/LocationsController.scala (hunk 1 above)
Comment (jamesgorrie) [reply onward/app/weather/controllers/LocationsController.scala:39]: Country / city are needed, and nearly always available. Region is way more Maybe like. It's just how my brain was thinking of it, and the order we use in the methods `(country: String, city: String, maybeRegion: Option[String])`.

**Comment by jamesgorrie** [comment]
> We could still add some tests, possibly at least for all the cities where we have an office.

We'd have to mock out the API for this. What I might want to test is the controller to explain the flow a little better. e.g. "when we have headers we do this, otherwise we do this".

File: onward/app/weather/controllers/LocationsController.scala [hunk 3]
```diff
@@ -34,44 +34,27 @@ class LocationsController(weatherApi: WeatherApi, val controllerComponents: Cont
...
       val maybeRegion = getEncodedHeader(RegionHeader).filter(_.nonEmpty)
-      val maybeCountry = getEncodedHeader(CountryHeader).filter(_.nonEmpty)
-
-      (maybeCity, maybeRegion, maybeCountry) match {
-        case (Some(city), Some(region), Some(country)) =>
-          CitiesLookUp.getLatitudeLongitude(CityRef.makeFixedCase(city, region, country)) match {
-            case Some(latitudeLongitude) =>
-              log.info(s"Matched $city, $region, $country to $latitudeLongitude")
 
-              weatherApi.getNearestCity(latitudeLongitude) map { location =>
```
Comment (ioannakok) [inline onward/app/weather/controllers/LocationsController.scala:47]: @jamesgorrie @mxdvl do you remember why removed the search by latitute / longitude?

cc @abeddow91 @cemms1

**Pull request merged**